mirgecom/euler.py

```diff
@@ -150,17 +181,26 @@ def inviscid_flux(discr, eos, q):
     r"""Compute the inviscid flux vectors from flow solution *q*.
 
     The inviscid fluxes are
-    :math:`(\rho\vec{V},(\rhoE+p)\vec{V},\rho(\vec{V}\otimes\vec{V})+p\mathbf{I})`
+    :math:`(\rho\vec{V},(\rho{E}+p)\vec{V},\rho(\vec{V}\otimes\vec{V})
+    +p\mathbf{I}, \rho{Y_s}\vec{V})`
     """
     dim = discr.dim
     cv = split_conserved(dim, q)
     p = eos.pressure(cv)
 
     mom = cv.momentum
+
+    massfrac = None
+    if cv.massfrac is not None:
+        # mass fractions require a reshape here to get the right numpy
+        # broadcast behavior. Reshaped to [numspecies x 1] object.
+        massfrac = mom * cv.massfrac.reshape(-1, 1) / scalar(cv.mass)
```

Comment:
```suggestion
        # massfrac shape: (numspecies, dim)
        massfrac = mom * cv.massfrac.reshape(-1, 1) / scalar(cv.mass)
```
(IMO knowledge of numpy broadcasting is assumed for someone reading this.)